starters/express-apollo-prisma/src/graphql/server-context/server-context-middleware-options.ts

```diff
@@ -1,21 +1,24 @@
-import { ServerContext, ServerContextDataSources } from './server-context';
+import { ServerContext } from './server-context';
 import { ExpressMiddlewareOptions } from '@apollo/server/express4';
 import { WithRequired } from '@apollo/utils.withrequired';
 import { TechnologyDataSource } from '../data-sources';
 import { PrismaClient } from '@prisma/client';
+import { RedisClient } from '../../redis';
 
 const prismaClient = new PrismaClient();
 
-const dataSources: ServerContextDataSources = {
-	technologyDataSource: new TechnologyDataSource(prismaClient),
-};
-
-export const serverContextMiddlewareOptions: WithRequired<
-	ExpressMiddlewareOptions<ServerContext>,
-	'context'
-> = {
+export const createServerContextMiddlewareOptions = (
+	redisClient?: RedisClient,
+	redisCacheTtlSeconds?: number
+): WithRequired<ExpressMiddlewareOptions<ServerContext>, 'context'> => ({
 	context: async ({ req }) => ({
-		dataSources,
+		dataSources: {
+			technologyDataSource: new TechnologyDataSource(
+				prismaClient,
+				redisClient,
+				redisCacheTtlSeconds
+			),
```

Comment:
Is there a way to auto import these? This would cause alot of boiler-plate for multiple tables for users